src/Host/Client/Impl/Host/RHost.cs

```diff
@@ -396,16 +396,16 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator, IRBlobSe
                     await _transport.CloseAsync();
                     token.Set();
                 } catch (OperationCanceledException) {
+                    token.Set();
                 } catch (MessageTransportException) {
+                    token.Set();
                 } finally {
                     token.Reset();
                 }
             }
 
             try {
-                if (_runTask.Status == TaskStatus.Running) {
```

Comment:
No, `WaitingForActivation` is a normal state for a task without action, which you get when you use `TaskCompletionSource` or get task from `async` method that contains custom awaiters. In this particular case, `_runTask` will be in `WaitingForActivation` state until `RunWorker` is completed, and then will be immediately switched to either `RanToCompletion` or  `Failed` or `Canceled`.